### ISSUE TITLE
SAS Token compatibility across SDKs versions (v12 - v7.5)

### ISSUE THREAD

**Issue by lemon-guy** [Azure/azure-sdk-for-cpp#3925]
**Query/Question**
I created a ticket here: [#3870 ](https://github.com/Azure/azure-sdk-for-cpp/issues/3870), as I thought Azure v12 API didn't work with Azure v7.5 generated SAS token. But in the end issue was on my end: I tried to .GetAccountInfo() while not having permissions for such operation.

This time though situation is reversed. I tried to call Azure v7.5 API with Azure v12 generated SAS token.
For example calling: azure::storage::cloud_block_blob::download_range_to_stream_async() will throw exception with such message:
`Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.`
Is such configuration supported? 

With v12 I generated this token:
`?se=2022-09-07T14:07:50Z&sig=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxHwEFu0bx5jVAD&sp=racwdl&spr=https&sr=c&sv=2020-10-02`

With v7.5 I generated this token:
`?sv=2019-12-12&sig=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxtYFC%2BIoSk53P2Yw%3D&spr=https&se=2022-08-05T18%3A02%3A58Z&sp=racwdl&sr=c`

The only thing that seems to differ is SignedVersion (sv). V7.5 is 2019-12-12, V12 is 2020-10-02. Is there a way to generate SAS token with older SignedVersion in V12? 

***Why is this not a Bug or a feature Request?***
It may be a bug, but perhaps I'm not using the SDK correctly?

**Setup (please complete the following information if applicable):**
 - OS: Debian10/CentOS7
 - IDE : Visual Code
 - Version of the Library used:
Azure Core 1.7.1
Azure Storage Common 12.2.4
Azure Storage Blobs 12.5.0
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Query Added
- [X] Setup information Added

**Comment by Jinming-Hu** [comment]
Can you share a request ID of the failed request? so that I can check server-side log.

**Comment by lemon-guy** [comment]
@Jinming-Hu Sure, here is a couple: 
- e20bd77f-501e-00a4-6c70-c4148d000000 
- e20bd76a-501e-00a4-5870-c4148d000000
They are very recent, I just sent them. My region is East US 2.

Two remarks:
- Client API is v12, but still failed (so I must have been not creating service tokens correctly. Although I don't understand how, as they look identical to what was generated in v7).
- I turned on "Diagnostic settings (classic)" for my Storage Account, but I don't see those failing request in logs. Is that expected?

**Comment by lemon-guy** [comment]
Ok, I think I found my issue. 
I was creating service SAS Token incorrectly. Namely I didn't set BlobContainerName to container name (I'm using Azure::Storage::Sas::BlobSasBuilder).
I stumbled upon .NET example, where that was shown. I would love to have more examples of C++ API, but it is what it is.

Anyway, compatibility between v12 and v7 SAS tokens works both ways. Just be careful whether you are using service or account SAS tokens. Thanks!
